CurvedSegment.py

```diff
@@ -128,32 +139,59 @@ def makeRibs(self, fp):
         else:
             ribs = makeRibsSameShape(fp, fp.Items, False, makeStartEnd)
             
-        if len(fp.Hullcurves) > 0:
-            self.rescaleRibs(fp, ribs)
+        self.rescaleRibs(fp, ribs)
             
         if fp.makeSurface or fp.makeSolid:
-            fp.Shape = CurvedShapes.makeSurfaceSolid(ribs, fp.makeSolid, maxDegree=fp.LoftMaxDegree)
+            fp.Shape = CurvedShapes.makeSurfaceSolid(ribs, fp.makeSolid, maxDegree=fp.LoftMaxDegree, maxLoftSize=fp.MaxLoftSize)
         else:
             fp.Shape = Part.makeCompound(ribs)          
         
         
     def rescaleRibs(self, fp, ribs):
-        if fp.makeSurface or fp.makeSolid:
+        if (fp.makeSurface or fp.makeSolid) and fp.Path is None and abs(fp.ActualTwist)<=epsilon:
             start = 1
             end = len(ribs) - 1
             items = fp.Items + 3
         else:   
             start = 0
             end = len(ribs) 
             items = fp.Items + 1
-            
+        
+        maxlen = 0   
+        edgelen = []
+        if fp.Path is not None:
+            edges = Part.__sortEdges__(fp.Path.Shape.Edges)
+            for edge in edges:
+                maxlen += edge.Length
+                edgelen.append(edge.Length)
+
+        bc0=fp.Shape1.Shape.Placement.Base
+        bc1=fp.Shape2.Shape.Placement.Base # makes rotating assymetric shapes easier - taking sketch origin into account
         for i in range(start, end):
-            d = CurvedShapes.distribute((i + 1) / items, fp.Distribution, fp.DistributionReverse)
```

Comment:
this was wrong anyway - since start would be 1 - i starts with 1 - no +1 needed or it goes beyond the end